File: course/exercise_lorenz_3var_part2/all_runs.m
% Script that will produce the figures of exercise 1
% Note your environment must consists of a configured OpenDA distro 
% since we are starting OpenDA from this script.
%
% Author Nils van Velzen

if (~ exist('enkf_results_std1_ens10'))     system('oda_run.sh EnkF_std1_ens10.oda');      end
if (~ exist('enkf_results_std10_ens10'))   system('oda_run.sh EnkF_std10_ens10.oda');    end
if (~ exist('enkf_results_std5_ens5'))   system('oda_run.sh EnkF_std5_ens5.oda');    end
if (~ exist('enkf_results_std5_ens10'))   system('oda_run.sh EnkF_std5_ens10.oda');    end
if (~ exist('enkf_results_std5_ens100'))   system('oda_run.sh EnkF_std5_ens100.oda');    end


figure(1);
[t,ens]=load_ensemble('enkf_results_std1_ens10');
[tobs,obs]=load_obs ('enkf_results_std1_ens10');
ens1=reshape(ens(1,:,:),size(ens,2),size(ens,3));
plot(t,ens1)
title('Lorenz model first variable (EnKF)')

disp('push a button to see mean and obs')
pause
plot(t,mean(ens1,2));
hold on
plot(tobs,obs,'*r');
hold off
title('Lorenz model mean and observations of first variable (EnKF)')

disp('Push a button to see the stdev=10')
pause

clear

figure(1);
[t,ens]=load_ensemble('enkf_results_std10_ens10');
[tobs,obs]=load_obs ('enkf_results_std10_ens10');
ens1=reshape(ens(1,:,:),size(ens,2),size(ens,3));
plot(t,ens1)
title('Lorenz model first variable, obs stdvar=10 (EnKF)')
disp('push a button to see mean and obs')
pause
plot(t,mean(ens1,2));
hold on
plot(tobs,obs,'*r');
hold off
title('Lorenz model mean and observations of first variable (EnKF)')

disp('Push a button to see various ensemble sizes with stdev=5')
pause


figure(1);
[tobs,obs]=load_obs ('enkf_results_std5_ens5');
[t,ens5]=load_ensemble('enkf_results_std5_ens5');
[t,ens10]=load_ensemble('enkf_results_std5_ens10');
[t,ens100]=load_ensemble('enkf_results_std5_ens100');

ens5=reshape(ens5(1,:,:),size(ens5,2),size(ens5,3));
ens10=reshape(ens10(1,:,:),size(ens10,2),size(ens10,3));
ens100=reshape(ens100(1,:,:),size(ens100,2),size(ens100,3));
m5=mean(ens5,2);
m10=mean(ens10,2);
m100=mean(ens100,2);


plot(t,m5, t, m10, t, m100);
title('Lorenz EnKF various ensemble sizes')
legend('n=5','n=10','n=100');

disp('Push a button to zoomin')
pause
axis([20 30 -20 20])



disp('Push a button to see the obs')
pause
%axis([0 30 -20 20])
hold on
plot(tobs,obs,'*r');
hold off

File: course/exercise_lorenz_3var_part2/all_runs.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
"""
Script that will produce the figures of exercise 2

@author: verlaanm
"""

#load numpy and matplotlib if needed
import numpy as np
import matplotlib.pyplot as plt
import ensemble

# laod data fomr enkf_results_estd1_ens10, collect ensemble in one 3d-array
# and compute ensemble mean of first [0] element in statevector (state=[x,y,z])
import enkf_results_std1_ens10 as std1
(t,enkf)=ensemble.reshape_ensemble(std1)
enkf1=enkf[:,0,:] #note we start counting at 0

#
#plot all ensemble members
#
fig1 = plt.figure()
plt.plot(t,enkf1)
plt.title("Lorenz model first variable (EnKF)")
plt.savefig("figure_1.png")

#
#plot ensemble mean and observations
#
fig2 = plt.figure()
plt.plot(t,np.mean(enkf1,1))
plt.plot(t,std1.obs[:,0],"r*")
plt.title("Lorenz model mean and observations of first variable (EnKF)")
plt.legend(("ensemble mean for x, std=1","observations"))
plt.savefig("figure_2.png")

#
#now for std=10
#
import enkf_results_std10_ens10 as std10
(t,enkf)=ensemble.reshape_ensemble(std10)
enkf10=enkf[:,0,:] #note we start counting at 0

#
#plot all ensemble members
#
fig3 = plt.figure()
plt.plot(t,enkf10)
plt.title("Lorenz model first variable (EnKF)")
plt.savefig("figure_3.png")

#
#plot ensemble mean and observations
#
fig4 = plt.figure()
plt.plot(t,np.mean(enkf10,1))
plt.plot(t,std10.obs[:,0],"r*")
plt.title("Lorenz model mean and observations of first variable (EnKF)")
plt.legend(("ensemble mean for x, std=10","observations"))
plt.savefig("figure_4.png")

#
# compare data for different ensemble sizes
#
import enkf_results_std5_ens5 as ens5
(t,ens5.ens)=ensemble.reshape_ensemble(ens5)
ens5.x=np.mean(ens5.ens[:,0,:],1)
import enkf_results_std5_ens10 as ens10
(t,ens10.ens)=ensemble.reshape_ensemble(ens10)
ens10.x=np.mean(ens10.ens[:,0,:],1)
import enkf_results_std5_ens100 as ens100
(t,ens100.ens)=ensemble.reshape_ensemble(ens100)
ens100.x=np.mean(ens100.ens[:,0,:],1)
#make plot
fig5 = plt.figure()
plt.plot(t,ens5.x,'k-')
plt.plot(t,ens10.x,'b-')
plt.plot(t,ens100.x,'g-')
plt.plot(t,ens10.obs[:,0],"r*")
plt.legend(("n=5","n=10","n=100","observations"))
plt.savefig("figure_5.png")

